corehq/apps/app_manager/suite_xml/post_process/instances.py

```diff
@@ -354,9 +356,13 @@ def search_input_instances(app, instance_name):
     try:
         _, query_datum_id = instance_name.split(':', 1)
         src = f'jr://instance/search-input/{query_datum_id}'
+        is_valid_instance_name = is_valid_results_instance_name(app, query_datum_id)
     except ValueError:
         src = 'jr://instance/search-input'  # legacy instance
-    return Instance(id=instance_name, src=src)
+        is_valid_instance_name = True
+    if is_valid_instance_name:
+        return Instance(id=instance_name, src=src)
+    return None
```

Comment:
This function reads a little weird - there are two fallback options, the legacy instance and `None`, and it's a bit difficult to tell which should be returned when.  Would it be fair to write this as:

```python
def search_input_instances(app, instance_name):
    if ":" not in instance_name:
        return Instance(id=instance_name, src='jr://instance/search-input')

    _, query_datum_id = instance_name.split(':', 1)
    if is_valid_results_instance_name(app, query_datum_id):
        src = f'jr://instance/search-input/{query_datum_id}'
        return Instance(id=instance_name, src=src)
    return None
```

Reply:
yes, [c5dba28](https://github.com/dimagi/commcare-hq/pull/33635/commits/c5dba28dbe1039bdec6a6e0cf9e1ec89bb255bbc)